storages/redis-storage/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+#![deny(clippy::str_to_string)]
+
+mod alter_table;
+mod index;
+mod metadata;
+mod transaction;
+
+use {
+    async_trait::async_trait,
+    futures::stream::iter,
+    gluesql_core::{
+        chrono::Utc,
+        data::{CustomFunction as StructCustomFunction, Key, Schema, Value},
+        error::{Error, Result},
+        store::{CustomFunction, CustomFunctionMut, DataRow, RowIter, Store, StoreMut},
+    },
+    redis::{Commands, Connection},
+    std::cell::RefCell,
+    std::collections::{BTreeMap, HashMap},
+};
+
+pub struct Item {
+    pub schema: Schema,
+}
+
+pub struct RedisStorage {
+    pub namespace: String,
+    pub items: HashMap<String, Item>,
+    pub conn: RefCell<Connection>,
+    pub functions: HashMap<String, StructCustomFunction>,
+}
```

Comment:
here `items` or `functions` fields might be considered as a cache but because it is redis storage. it would be too much for the start.
let's make the storage implementation simple. it would be enough to only have `namespace` and `conn` fields.
in the case of `items` we can always store & fetch from redis rather than cloning it in here.

and.. `functions`, you may not be able to implement custom function trait for now. (it is because of the limitation of the current interface)

Reply:
Got it. I will remove items and functions fields.

Reply:
Done